src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -75,18 +80,34 @@
     private final int lineNumber;
     /** The column number of the element for which the query should be generated. */
     private final int columnNumber;
+    /** The {@code FileText} object, representing content of the file. */
+    private final FileText fileText;
+    /** The distance between tab stop position. */
+    private final int tabWidth;
 
     /**
      * Creates a new {@code XpathQueryGenerator} instance.
      *
      * @param rootAst root ast
      * @param lineNumber line number of the element for which the query should be generated
      * @param columnNumber column number of the element for which the query should be generated
+     * @param fileName path to the file
+     * @param tabWidth distance between tab stop position
      */
-    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber) {
+    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber,
+                               String fileName, int tabWidth) {
```

Comment:
@MEZk any reason why we need file name versus file or file text?
Event `AstTreeStringPrinter` and `DetailNodeTreeStringPrinter` take `File` as an argument.

Reply:
@rnveach 
No. The generator will be invoked from Main. Main has the File instance in config, so it can pass File to the generator.

@timurt 
Please, fix.

Reply:
done